src/gluonts/dataset/split/splitter.py

```diff
@@ -250,13 +249,24 @@ def rolling_split(
                 )
                 prediction_length = getattr(self, "prediction_length")
 
-                assert (
-                    train.end + train.end.freq * prediction_length <= test.end
+                self._check_split_length(
+                    train.end, test.end, prediction_length
                 )
                 split._add_test_slice(test)
 
         return split
 
+    @staticmethod
+    def _check_split_length(
+        train_end: pd.Period, test_end: pd.Period, prediction_length: int
+    ) -> None:
+        msg = (
+            "Number of left over values is too small to create the test"
+            + " window. Either the timeseries is too short or"
+            + " ``split_offset`` or ``split_date`` is not set properly."
```

Comment:
I think this goes into details of the subclasses, which is not ideal. Maybe just saying “Not enough observations after the split point to construct the test instance; consider using longer time series, or splitting at an earlier point”